src/Components/__test__/pagination.test.tsx

```diff
@@ -0,0 +1,70 @@
+import { render } from '@testing-library/react'
+import '@testing-library/jest-dom'
+import Pagination from '../pagination'
+
+jest.mock('next/router', () => ({
+  useRouter: jest.fn().mockReturnValue({
+    pathname: '/',
+    query: { language: 'javascript' },
+  }),
+}))
+
+describe('Paginationのテスト', () => {
+  test('Paginationのページ数が6以下の場合にlinkの数がページと同じように表示されているかどうか', () => {
+    const totalCount = 100
+    const perPage = 50
+    const totalPage = Math.ceil(totalCount / perPage) // 2
+    const { getAllByRole } = render(
+      <Pagination currentPage={1} totalCount={totalCount} perPage={perPage} />,
+    )
+    expect(getAllByRole('link')).toHaveLength(totalPage)
+  })
+  test('Paginationのページ数が7以上の場合に「...」が表示されているかどうか', () => {
+    const totalCount = 350
+    const perPage = 50
+    const totalPage = Math.ceil(totalCount / perPage) // 7
+    const { getByText } = render(
+      <Pagination currentPage={1} totalCount={totalCount} perPage={perPage} />,
+    )
+    expect(getByText('...')).toBeInTheDocument()
+  })
+  test('Paginationのページ数が7以上の場合にlinkの数が5つ表示されているかどうか', () => {
+    const totalCount = 301
+    const perPage = 50
+    const totalPage = Math.ceil(totalCount / perPage) // 7
+    const { getAllByRole } = render(
+      <Pagination currentPage={1} totalCount={totalCount} perPage={perPage} />,
+    )
+    expect(getAllByRole('link')).toHaveLength(5)
+  })
```

Comment:
`ページ数が7以上の場合`のテストで、`currentPage `が`urrentPage <= 3`以外の2パターンについても網羅しておいてください🙏

```suggestion
//例
test('Paginationのページ数が7以上の場合かつ現在ページが4のときに左右に「...」が表示されているかどうか', () => {
//テスト
})
test('Paginationのページ数が7以上の場合かつ現在ページが4のときにlinkの数が5つ表示されているかどうか', () => {
//テスト
})
test('Paginationのページ数が7以上の場合かつ現在ページが最終ページの2ページ前のときに左に「...」が表示されているかどうか', () => {
//テスト
})
test('Paginationのページ数が7以上の場合かつ現在ページが最終ページの2ページ前のときにlinkの数が5つ表示されているかどうか', () => {
//テスト
})
```